govcd/edgegateway.go

```diff
@@ -634,3 +640,53 @@ func (eGW *EdgeGateway) RemoveIpsecVPN() (Task, error) {
 	}
 	return eGW.AddIpsecVPN(ipsecVPNConfig)
 }
+
+// Deletes the edge gateway, returning an error if the vCD call fails.
+// https://code.vmware.com/apis/442/vcloud-director/doc/doc/operations/DELETE-EdgeGateway.html
+func (egw *EdgeGateway) Delete(force bool, recursive bool) (Task, error) {
```

Comment:
According our previous discussions and decisions I would make Delete sync method which returns only error and DeleteTask or similar name which return async Task to handle by user themself. Edge gtw only handle one action that's why I propose to do that.

Reply:
done. We now have two functions: `Delete` returns an error, and `DeleteTask` returns task + error.